### DISCUSSION TITLE
Failure to use GlobalTables.

### DISCUSSION THREAD

**Discussion by hardincm2** [LGouellec/streamiz#129] [General]
I have a fairly simple main program that fails pretty silently while trying to initialize the global state store (no further details). This works perfectly fine with regular kTables and streams.

```
fail: Streamiz.Kafka.Net.Processors.StreamStateManager[0]
      Global thread has died. The instance will be in error state and should be closed
warn: Streamiz.Kafka.Net.Processors.GlobalStreamThread[0]
      global-stream-thread cam-test-app-930d0e77-03a9-4f59-b213-c3982c8678ed-GlobalStreamThread Error happened during initialization of the global state store; this thread has shutdown
```

Code
```
    public static async Task<int> Main(string[] args)
    {
        var config = new StreamConfig
        {
            ApplicationId = "test-app",
            BootstrapServers = Environment.GetEnvironmentVariable("Secret1"),
            SslCaLocation = "./my-ca.crt",
            SslCertificatePem = Environment.GetEnvironmentVariable("Secret2"),
            SslKeyPem = Environment.GetEnvironmentVariable("Secret3"),
            SecurityProtocol = SecurityProtocol.Ssl,
            AutoOffsetReset = AutoOffsetReset.Earliest,
        };

        StreamBuilder builder = new StreamBuilder();

        var table = builder.GlobalTable<string, Data>("topic", new StringSerDes(), new JsonSerDes<Data>());

        Topology t = builder.Build();

        KafkaStream stream = new KafkaStream(t, config);
        await stream.StartAsync();

        return 0;
    }
```

Running .Net 6 and Ubuntu 20.04. Any thoughts on what may be going on here? Next step is just to pull source and try to step through the code to figure what's breaking.

**Comment by LGouellec** [comment]
Could you activate debugging log ? 
Example here : https://github.com/LGouellec/kafka-streams-dotnet/blob/b26e90b5ddd237720fea17e921eb44484f15a536/samples/sample-stream/Program.cs#L37

And activate debugging log on librdkafka ?
Documentation (debug confit key) here : https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md

Example : https://github.com/LGouellec/kafka-streams-dotnet/blob/b26e90b5ddd237720fea17e921eb44484f15a536/core/StreamConfig.cs#L1266

**Reply by hardincm2** [reply]
Only getting two more lines out of it with debugging pieces enabled. Doesn't seem particularly insightful
```
dbug: Streamiz.Kafka.Net.Processors.DefaultTopicManager[0]
      Starting to apply internal topics in topic manager (try: 1, max retry : 10).
dbug: Streamiz.Kafka.Net.Processors.DefaultTopicManager[0]
      Complete to apply internal topics in topic manager
```

**Reply by LGouellec** [reply]
I reproduce the issue, fix it and track into this [PR](https://github.com/LGouellec/kafka-streams-dotnet/pull/131).

Will be available in 1.3.0 release soon
